### PR TITLE
237 v2 App Projects View

### LINKED ISSUE #237: Project Details

### PR DESCRIPTION
**Pull request by cherry-john**
Add Projects to v2 app
closes #237 

## Version:
v2

## Checklist:

- [ ] Updates documentation (if so - is it in this PR, or on the v2 branch?)
- [x] Tested locally

### CODE REVIEW

**Comment by Robert-Watts** [comment]
image


- [ ] Message when there are no projects to display

**Comment by Robert-Watts** [comment]
> image
> 
> * [ ]  Message when there are no projects to display

Sorry had not realised this was a draft PR, my bad

**Comment by Jbithell** [comment]
@cherry-john despite me doing a succesful merge just before and then running the lint in this branch, there are now a few more conflicts. I am not familiar enough with this branch to really know whats' going on here but hopefully you'll be able to sort them relatively easily now we've done this merging

**Comment by cherry-john** [comment]
More project work to come, this is the basic project view sorted

File: app/src/contexts/Context.tsx [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
-            </AssetTypeProvider>
-        </>
-    )
-}
\ No newline at end of file
+  return (
+    <>
+      <AssetTypeProvider>
+        <ProjectProvider>
+          <ProjectDataProvider>{props.children}</ProjectDataProvider>
```
Comment (Robert-Watts) [inline app/src/contexts/Context.tsx:15]: This is going to get quite long/deep lol, but don't do anything about it on this PR, I think I have an idea I might play with.

File: app/src/pages/projects/Project.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+              projectData.project.crewAssignments.map((item: any) => {
+                return (
+                  <IonItem key={item.crewAssignments_id}>
+                    <IonLabel>
+                      {item.users_name1} {item.users_name2}
+                      <p>{item.crewAssignments_role}</p>
+                    </IonLabel>
+                    {
+                      item.crewAssignments_comment /* && (
+                        TODO: uncomment this when updated to Ionic 6
```
Comment (Robert-Watts) [inline app/src/pages/projects/Project.tsx:168]: Is this an issue so we do not forget?

File: app/src/pages/projects/ProjectAssets.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    };
+    price: {
+      amount: string;
+      currency: string;
+    };
+    mass: number;
+  };
+}
+
+const ProjectAssets = () => {
```
Comment (Robert-Watts) [inline app/src/pages/projects/ProjectAssets.tsx:42]: Comment like ProjectList.tsx

File: app/src/pages/projects/Project.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  IonTitle,
+} from "@ionic/react";
+import { useContext, useEffect } from "react";
+import { useParams } from "react-router";
+import { ProjectDataContext } from "../../contexts/project/ProjectDataContext";
+import Page from "../../components/Page";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { baseURL } from "../../utilities/Api";
+
+const Project = () => {
```
Comment (Robert-Watts) [inline app/src/pages/projects/Project.tsx:26]: Comment like ProjectList.tsx

File: app/src/type.d.ts [hunk 5]
```diff
@@ -2,87 +2,103 @@
...
+/* Project Object */
+interface IProject {
+  projects_id: number;
+  projects_name: string;
+  clients_name: string;
+  projects_manager: number;
+  thisProjectManager: boolean;
+}
+
+interface IProjectData {
```
Comment (Robert-Watts) [inline app/src/type.d.ts:99]: Comment like `IProject ` with what this is for

**Review by Robert-Watts — COMMENTED** [review]
Looks good, a few comments but nothing major. I have not tested this locally so as long as it works on your PC I am happy.

File: app/src/pages/projects/Project.tsx (hunk 2 above)
Comment (cherry-john) [reply app/src/pages/projects/Project.tsx:168]: #273

**Review by Robert-Watts — DISMISSED** [review]

File: adam-rms.code-workspace [hunk 6]
```diff
@@ -5,7 +5,7 @@
 		}
 	],
 	"settings": {
-		"editor.tabSize": 4,
+		"editor.tabSize": 2,
```
Comment (Jbithell) [inline adam-rms.code-workspace:8]: https://google.github.io/styleguide/jsguide.html#formatting-indent

Is the above link conflicting with eslint? (https://github.com/google/gts/issues/626)

File: adam-rms.code-workspace [hunk 7]
```diff
@@ -16,6 +16,7 @@
 		"editor.codeActionsOnSave": {
 			"source.fixAll.eslint": true
 		},
+		"eslint.validate": ["javascript"],
```
Comment (Jbithell) [inline adam-rms.code-workspace:19]: should this be typescript?

File: app/src/utilities/Formatters.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ * @param currency 3 digit currency string
+ * @param amount amount without decimal place
+ * @returns formatted string
+ */
+export const MoneyFormatter = (currency: string, amount: string) => {
+  const formatter = new Intl.NumberFormat(undefined, {
+    style: "currency",
+    currency: currency,
+  });
+  return formatter.format(parseInt(amount) / 100);
```
Comment (Jbithell) [inline app/src/utilities/Formatters.ts:12]: Should this be /100? I think it varies by currency

**Review by Jbithell — CHANGES_REQUESTED** [review]

File: adam-rms.code-workspace (hunk 6 above)
Comment (cherry-john) [reply adam-rms.code-workspace:8]: yep, although i thought i'd only changed this on my local config.

Comment (Jbithell) [reply adam-rms.code-workspace:8]: @cherry-john shall we revert this then?

**Comment by Jbithell** [comment]
@cherry-john can you fix eslint failures to allow me to approve & merge?

File: app/src/utilities/Formatters.ts (hunk 8 above)
Comment (cherry-john) [reply app/src/utilities/Formatters.ts:12]: ah good point, I need to use a currency library

File: adam-rms.code-workspace (hunk 6 above)
Comment (cherry-john) [reply adam-rms.code-workspace:8]: makes most sense, will do in current changes

File: adam-rms.code-workspace (hunk 7 above)
Comment (cherry-john) [reply adam-rms.code-workspace:19]: Yes, have updated accordingly

File: app/src/utilities/Formatters.ts (hunk 8 above)
Comment (cherry-john) [reply app/src/utilities/Formatters.ts:12]: I'm going to make this an issue today, and fix soon, rather than in this commit

**Review by Jbithell — APPROVED** [review]

**Pull request merged**